### PR TITLE
Enabled database to store notes

### LINKED ISSUE #5: Use a database to store the notes
Right now notes are saved in text file, use a database for the purpose

### PR DESCRIPTION
**Pull request by PiyushDhamane**
Fixed #5 

Initially, the notes were stored in a local text file. But now, they'll be stored in MongoDB. 
Used mongoose to give a proper structure to the data being stored. 
Used npm install for installing the required modules. Hence other modules were updated as well. So the commit might seem a bit messy at first look, but I request you to try running it in the local system once. The code isn't messy at all and runs properly.

### CODE REVIEW

**Comment by PiyushDhamane** [comment]
@im-prakher please check this

File: notes-app/note.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+var Note = module.exports = mongoose.model('Note', NoteSchema);
+
+module.exports.getNoteByTitle = function(title, callback){
+	var query = {title: title};
+	Note.findOne(query, callback);
+}
+
+
+module.exports.createNote = function(newNote, callback){
```
Comment (im-prakher) [inline notes-app/note.js:25]: Can you just make sure that the notes don't have the same title? Perhaps, check for that in the database and then create the note.

**Review by im-prakher — COMMENTED** [review]
Thanks for the good work!

**Comment by PiyushDhamane** [comment]
@im-prakher done

**Comment by im-prakher** [comment]
Looks good to me!

**Pull request merged**